src/cce_rpc/engine.proto

```diff
@@ -428,14 +428,14 @@ message ScheduleDowntimeIdentifier {
   string host_group_name = 2;
   string service_group_name = 3;
   string service_desc = 4;
-  uint32 start = 5;
-  uint32 end = 6;
-  bool fixed = 7;
-  uint32 triggered_by = 8;
-  uint32 duration = 9;
+  google.protobuf.UInt32Value start = 5;
+  google.protobuf.UInt32Value end = 6;
+  google.protobuf.BoolValue fixed = 7;
+  google.protobuf.UInt32Value triggered_by = 8;
+  google.protobuf.UInt32Value duration = 9;
   string author = 10;
   string comment_data  = 11;
-  uint32 entry_time = 12;
+  google.protobuf.UInt32Value entry_time = 12;
```

Comment:
You seem to want those values. google.protobuf...Value gives the possibility to omit a value.
So I think you should keep what you did before:
uint32 start = 5;
uint32 end = 6;
etc...
In case of a value not filled, it will be considered as empty or null. This is told by the protobuf v3 specs. So you can then check values, and don't have to bother if people forgot to fill a value.